File: src/main/java/com/fish/service/label/LabelServiceImpl.java
package com.fish.service.label;

import com.fish.common.Result;
import com.fish.entity.pojo.Label;
import com.fish.exception.ServiceException;
import com.fish.exception.ServiceExceptionEnum;
import com.fish.mapper.LabelMapper;
import com.fish.utils.ResultUtil;
import com.mybatisflex.core.query.QueryWrapper;
import com.mybatisflex.spring.service.impl.ServiceImpl;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.ArrayList;
import java.util.Objects;

import static com.fish.entity.pojo.table.LabelTableDef.LABEL;

@Service
public class LabelServiceImpl extends ServiceImpl<LabelMapper, Label> implements LabelService {
    @Transactional
    @Override
    public Result<?> addLabel(Label label) {
        int i = mapper.insert(label);
        if (i > 0)
            return ResultUtil.success();
        throw new ServiceException(ServiceExceptionEnum.OPERATE_ERROR);
    }
    @Transactional
    @Override
    public Result<?> addLabelBatch(ArrayList<Label> labels) {
        if (labels.isEmpty())
            throw new ServiceException(ServiceExceptionEnum.KEY_ARGUMENT_NOT_INPUT);
        int i = mapper.insertBatch(labels);
        if (i > 0)
            return ResultUtil.success();
        throw new ServiceException(ServiceExceptionEnum.OPERATE_ERROR);
    }
    @Override
    public Result<ArrayList<Label>> getLabels() {
        ArrayList<Label> labels = (ArrayList<Label>) mapper.selectAll();
        return ResultUtil.success(labels);
    }
    @Override
    public Result<Label> getLabel(Long labelId) {
        Label label = mapper.selectOneById(labelId);
        if (!Objects.isNull(label))
            return ResultUtil.success(label);
        throw new ServiceException(ServiceExceptionEnum.NOT_FOUND);
    }
    @Transactional
    @Override
    public Result<?> updateLabel(Label label) {
        if (Objects.isNull(label.getLabelId()))
            throw new ServiceException(ServiceExceptionEnum.KEY_ARGUMENT_NOT_INPUT);
        QueryWrapper wrapper = QueryWrapper.create().select().from(LABEL).where(LABEL.LABEL_ID.eq(label.getLabelId()));
        int i = mapper.updateByQuery(label, wrapper);
        if (i > 0)
            return ResultUtil.success();
        throw new ServiceException(ServiceExceptionEnum.OPERATE_ERROR);
    }
    @Transactional
    @Override
    public Result<?> deleteLabel(Long labelId) {
        int i = mapper.deleteById(labelId);
        if (i > 0)
            return ResultUtil.success();
        throw new ServiceException(ServiceExceptionEnum.OPERATE_ERROR);
    }
}

File: src/main/kotlin/com/fish/service/label/impl/LabelServiceImpl.kt
package com.fish.service.label.impl

import com.fish.common.Result
import com.fish.entity.pojo.Label
import com.fish.entity.pojo.table.LabelTableDef
import com.fish.exception.ServiceException
import com.fish.exception.ServiceExceptionEnum
import com.fish.mapper.LabelMapper
import com.fish.service.label.LabelService
import com.fish.utils.ResultUtil.success
import com.mybatisflex.core.query.QueryWrapper
import com.mybatisflex.spring.service.impl.ServiceImpl
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.util.*
import kotlin.collections.ArrayList

@Service
class LabelServiceImpl : ServiceImpl<LabelMapper, Label>(), LabelService {
    @Transactional
    override fun addLabel(label: Label): Result<*> {
        val i = mapper!!.insert(label)
        if (i > 0) return success<Any>()
        throw ServiceException(ServiceExceptionEnum.OPERATE_ERROR)
    }

    @Transactional
    override fun addLabelBatch(labels: ArrayList<Label>): Result<*> {
        if (labels.isEmpty()) throw ServiceException(ServiceExceptionEnum.KEY_ARGUMENT_NOT_INPUT)
        val i = mapper!!.insertBatch(labels)
        if (i > 0) return success<Any>()
        throw ServiceException(ServiceExceptionEnum.OPERATE_ERROR)
    }

    override fun getLabels(): Result<ArrayList<Label>> {
        val labels = mapper!!.selectAll() as ArrayList<Label>
        return success(labels)
    }

    override fun getLabel(labelId: Long): Result<Label> {
        val label = mapper!!.selectOneById(labelId)
        if (!Objects.isNull(label))
            return success(label!!)
        throw ServiceException(ServiceExceptionEnum.NOT_FOUND)
    }

    @Transactional
    override fun updateLabel(label: Label): Result<*> {
        if (Objects.isNull(label.labelId)) throw ServiceException(ServiceExceptionEnum.KEY_ARGUMENT_NOT_INPUT)
        val wrapper = QueryWrapper.create().select().from(LabelTableDef.LABEL)
            .where(LabelTableDef.LABEL.LABEL_ID.eq(label.labelId))
        val i = mapper!!.updateByQuery(label, wrapper)
        if (i > 0) return success<Any>()
        throw ServiceException(ServiceExceptionEnum.OPERATE_ERROR)
    }

    @Transactional
    override fun deleteLabel(labelId: Long): Result<*> {
        val i = mapper!!.deleteById(labelId)
        if (i > 0) return success<Any>()
        throw ServiceException(ServiceExceptionEnum.OPERATE_ERROR)
    }
}
